src/content/en/fundamentals/performance/lazy-loading-guidance/images-and-video/index.md

```diff
@@ -0,0 +1,699 @@
+      entries.forEach(function(video) {
+        if (video.isIntersecting) {
+          for (var source in video.target.children) {
+            var videoSource = video.target.children[source];
+            if (typeof videoSource.tagName === "string" && videoSource.tagName === "SOURCE") {
+              videoSource.src = videoSource.dataset.src;
+            }
+          }
+
+          video.target.load();
+          video.target.classList.remove("lazy");
+          lazyVideoObserver.unobserve(video.target);
+        }
+      });
+    });
+
+    lazyVideos.forEach(function(lazyVideo) {
+      lazyVideoObserver.observe(lazyVideo);
+    });
+  }
+});
+```
+
+When we lazy load a `<video>` element, we need to iterate through all of the child
+`<source>` elements and flip their `data-src` attributes to `src` attributes. Once
+we've done that, we need to trigger loading of the video by calling the
+element's `load` method, after which the media will begin playing automatically
+per the `autoplay` attribute.
+
+Using this method, we have a video solution that emulates animated GIF behavior,
+but doesn't incur the same intensive data usage as animated GIFs do, and we get
+to lazy load that content.
+
+## Lazy loading libraries
+
+If you're not so concerned about _how_ lazy loading works under the hood and just
+want to pick a library and go (and there's no shame in that!), there's plenty of
+options to choose from. Many libraries use a markup pattern similar to the ones
+demonstrated in this guide. Here are some lazy loading libraries you may find
+useful:
+
+- [lazysizes](https://github.com/aFarkas/lazysizes) is a full-featured lazy
+loading library that lazy loads images and iframes. The pattern it uses is quite
+similar to the code examples shown here in that it automatically binds to a
+`lazyload` class on `<img>` elements, and requires you to specify image URLs in
+`data-src` and/or `data-srcset` attributes, the contents of which are swapped
+into `src` and/or `srcset` attributes, respectively. It uses intersection
+observer (which you can polyfill), and can be extended with [a number of
+plugins](https://github.com/aFarkas/lazysizes#available-plugins-in-this-repo) to
+do things like lazy load video.
+- [lozad.js](https://github.com/ApoorvSaxena/lozad.js) is a super lightweight
+option that uses intersection observer only. As such, it's highly performant,
+but will need to be polyfilled before you can use it on older browsers.
+- [blazy](https://github.com/dinbror/blazy) is another such option that bills
+itself as a lightweight lazy loader (weighing in at 1.4 KB). As with lazysizes,
+it doesn't need any third party utilities to load, and works for IE7+.
+Unfortunately, it doesn't use intersection observer.
+- [yall.js](https://github.com/malchata/yall.js) is a library I wrote that uses
+IntersectionObserver and falls back to event handlers. It's compatible with IE11
+and major browsers.
```

Comment:
I also just came across https://github.com/jasonslyvia/react-lazyload which appears popular in the React community. @StephenFluin is there a canonical library Angular recommends for lazy loading media right now?